src/components/molecules/TableHeader/TableHeader.tsx

```diff
@@ -138,60 +149,107 @@ const TableHeader: React.FC<TableHeaderProps> = ({
-            {tableTitle}
-
-            {tableCapacity && (
-              <span className="private-table-seats-left">
-                {numberOfSeatsLeft} seats left
-              </span>
-            )}
-          </div>
-          {tableOfUser && tableOfUser.subtitle && (
-            <div className="private-table-subtitle">{tableOfUser.subtitle}</div>
-          )}
-        </div>
-        <div className="lock-button-container">
+    <div className="row table-header">
+      <div className="table-header__leave-table">
+        <button
+          type="button"
+          title={"Leave " + seatedAtTable}
+          className="table-header__leave-button"
+          id="leave-seat"
+          onClick={leaveSeat}
+        >
           <FontAwesomeIcon
-            className="lock-table-icon"
-            icon={isCurrentTableLocked ? faLock : faLockOpen}
-            size="sm"
+            className="table-header__leave-button--icon"
+            icon={faChevronLeft}
+            size="xs"
           />
-          <div className="lock-table-checbox-indication">
-            {isCurrentTableLocked ? "Table Locked" : "Lock Table"}
+          Leave table
+        </button>
+      </div>
+
+      <div className="table-header__topic-info">
+        <div className="row table-header__topic">
+          <div>{tableTitle}</div>
+
+          <div className="table-header__edit-topic-button" onClick={show}>
+            <FontAwesomeIcon icon={faPen} />
           </div>
-          <label className="switch">
-            <input
-              type="checkbox"
-              className="switch-hidden-input"
-              checked={!!isCurrentTableLocked}
-              onChange={toggleIsCurrentTableLocked}
-            />
-            <span className="slider" />
-          </label>
         </div>
+
+        {tableCapacity && (
+          <span className="table-header__seats-left">
+            {numberOfSeatsLeft} seats left
+          </span>
+        )}
+
+        {tableOfUser && tableOfUser.subtitle && (
+          <label>{tableOfUser.subtitle}</label>
+        )}
+      </div>
+
+      <div className="table-header__lock-button">
+        <FontAwesomeIcon
+          className="table-header__lock-button--icon"
+          icon={isCurrentTableLocked ? faLock : faLockOpen}
+          size="sm"
+        />
+        <div className="table-header__lock-indication">
+          {isCurrentTableLocked ? "Table Locked" : "Lock Table"}
+        </div>
+        <label className="switch table-header__lock-toggle">
+          <input
+            type="checkbox"
+            className="switch-hidden-input"
+            checked={!!isCurrentTableLocked}
+            onChange={toggleIsCurrentTableLocked}
+          />
+          <span className="slider" />
+        </label>
       </div>
+
+      <EditTableTitleModal
+        isShown={isShown}
+        title={tableTitle}
+        error={error}
+        onHide={hide}
+        onCancel={hide}
+        onSave={updateTableTitle}
```

Comment:
Since `updateTableTitle` exists here seemingly only to be passed into this modal, would it make more sense for it to be defined within `EditTableTitleModal` itself? Seems like an implementation detail 'leaking out' of the scope of that component.

Reply:
I wasn't sure on how to proceed about this as well. Moving this inside will encapsulate it but force me to pass `tables` and `tableOfUser` as props. My initial thought was I'd rather pass 1 prop instead of 2. But I'm very passive about this. It's your call on which one is better and correct.

Reply:
As a general high level rule, I have no concerns about passing multiple props when it makes sense to, props are cheap/easy.

While I haven't looked deeply at the code, I can't imagine a reason to be concerned about passing `tables` and `tableOfUser` as props. Are there any aspects of that (aside from number of props) that come to mind as a concern for you? If not, I say that's probably a better approach.

Reply:
>  If not, I say that's probably a better approach.

In that case, I'll change it and proceed as you suggested.

Reply:
`onSave` callback has been removed in: https://github.com/sparkletown/sparkle/pull/1377/commits/4b7532e6f02068126c212a6f0dab1714aaaad55a

Reply:
The whole logic is inside the component now.